platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeDeclarationIntegrationTest.groovy

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.declarativedsl.settings
+
+import groovy.test.NotYetImplemented
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.plugin.PluginBuilder
+import org.gradle.test.fixtures.server.http.MavenHttpPluginRepository
+import org.junit.Rule
+
+class SoftwareTypeDeclarationIntegrationTest extends AbstractIntegrationSpec {
+    @Rule
+    MavenHttpPluginRepository pluginPortal = MavenHttpPluginRepository.asGradlePluginPortal(executer, mavenRepo)
+
+    @Rule
+    MavenHttpPluginRepository mavenHttpRepo = new MavenHttpPluginRepository(mavenRepo)
+
+    def 'can declare and configure a custom software type from included build'() {
+        given:
+        withSoftwareTypePlugins().prepareToExecute()
+
+        file("settings.gradle.something") << """
+            pluginManagement {
+                includeBuild("plugins")
+            }
+            plugins {
+                id("com.example.test-software-type")
+            }
+        """
+
+        file("build.gradle.something") << declarativeScriptThatConfiguresSoftwareType
+
+        when:
+        run(":printConfiguration")
+
+        then:
+        assertThatDeclaredValuesAreSetProperly()
+    }
+
+    def 'can declare and configure a custom software type from published plugin'() {
+        given:
+        def pluginBuilder = withSoftwareTypePlugins()
+        pluginBuilder.publishAs("com", "example", "1.0", pluginPortal, createExecuter()).allowAll()
+
+        file("settings.gradle.something") << """
+            plugins {
+                id("com.example.test-software-type") version("1.0")
```

Comment:
💭Feel free to ignore it in this PR, but I would suggest using the dot notation for `.version(...)` instead of the infix notation. The fact that it works is a historical coincidence. :) Eventually we should drop the infix  notation in Declarative DSL.

Reply:
I went ahead and changed this while I was in here.